Dockerfile

```diff
@@ -1 +1,28 @@
+ARG GOLANG_VERSION=1.19
+ARG MENDER_CLI_VERSION=master
+ARG MENDER_ARTIFACT_VERSION=master
+
+FROM golang:${GOLANG_VERSION} as cli-builder
+WORKDIR /go/src/github.com/mendersoftware/mender-cli
+ARG MENDER_CLI_VERSION
+RUN git clone -b $MENDER_CLI_VERSION https://github.com/mendersoftware/mender-cli.git . && \
+    make get-deps && \
+    make build
+
+FROM golang:${GOLANG_VERSION} as artifact-builder
+WORKDIR /go/src/github.com/mendersoftware/mender-artifact
+ARG MENDER_ARTIFACT_VERSION
+RUN git clone -b $MENDER_ARTIFACT_VERSION https://github.com/mendersoftware/mender-artifact.git . && \
+    apt-get update && apt-get install -y $(cat deb-requirements.txt) && \
+    make build
+
 FROM alpine
+COPY --from=cli-builder /go/src/github.com/mendersoftware/mender-cli /usr/bin/
+COPY --from=artifact-builder /go/src/github.com/mendersoftware/mender-artifact/mender-artifact /usr/bin/
+ARG MENDER_ARTIFACT_VERSION
+# The --mount instead of COPY is used to decrease the image layer size. Requires: 'export DOCKER_BUILDKIT=1'
+RUN --mount=type=bind,source=apk-requirements.txt,target=/apk-requirements.txt \
+    wget https://raw.githubusercontent.com/mendersoftware/mender-artifact/${MENDER_ARTIFACT_VERSION}/apk-run-requirements.txt && \
+    apk update && \
```

Comment:
With the `apk --no-cache` in the next line you do not need the update here, since the `no-cache` flag will get the latest package index right away and take the package from there...
But it also is not that much of extra build time with this includes, so feel free to merge away 🙂

Reply:
Impressive immersion in details 👍